modules/dhcp_common/isc/omapi_provider.rb

```diff
@@ -195,22 +194,22 @@ def poap_options_statements(options)
       statements
     end
 
-    def bootServer server
+    def bootServer(server)
+      ip2hex(validate_ip(server))
+    rescue
       begin
-        ns = ip2hex validate_ip(server)
-      rescue
-        begin
-          ns = ip2hex Resolv.new.getaddress(server)
-        rescue
-          logger.warn "Failed to resolve IP address for #{server}"
-          ns = "\\\"#{server}\\\""
-        end
+        logger.info "Next-server option not IPv4, trying to resolve '#{server}'"
+        ip2hex(dns_resolv.getaddress(server))
+      rescue Resolv::ResolvError => e
+        logger.warn "Unable to resolve record, using '#{server}' as next-server"
```

Comment:
Nit: this is not entirely true anymore: it can be option-150. On the other hand, we use the `nextServer` parameter so I'm fine either way.

Reply:
Improved the warning message a bit.